py/runtime.c

```
@@ -1593,8 +1606,28 @@ mp_obj_t mp_import_from(mp_obj_t module, qstr name) {
 void mp_import_all(mp_obj_t module) {
     DEBUG_printf("import all %p\n", module);
 
-    // TODO: Support __all__
     mp_map_t *map = &mp_obj_module_get_globals(module)->map;
+
+    #if MICROPY_ENABLE_MODULE_ALL
+    mp_map_elem_t *elem = mp_map_lookup(map, MP_OBJ_NEW_QSTR(MP_QSTR___all__), MP_MAP_LOOKUP);
+    if (elem != NULL && mp_obj_is_type(elem->value, &mp_type_list)) {
```

Would it be OK to just let it throw an exception later if the type is wrong?

---

`mp_obj_is_type()` was necessary as I was using `mp_obj_list_get()` afterwards, which does not perform any type checking. But as I have now switched to `mp_obj_get_array()` as per your next comment, this check can indeed be removed,